statsmodels/distributions/discrete.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+from scipy.stats import rv_discrete
+from scipy.special import gammaln
+
+
+class genpoisson_p_gen(rv_discrete):
+    '''A generalized Poisson discrete random variable
+    '''
+    def _argcheck(self, mu, alpha, p):
+        return True
+
+
+    def _logpmf(self, x, mu, alpha, p):
+        mu_p = mu ** (p - 1)
+        a1 = 1 + alpha * mu_p
+        a2 = mu + (a1 - 1) * x
```

Comment:
I think we need to clip a2 to be nonnegative, e.g. np.clip or np.maximum with lower bound 0.
if a2 = 0, then logpmf =-np.inf and pmf=0